utils/slice/slices.go

```diff
@@ -0,0 +1,203 @@
+// Copyright © 2022 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package slice
```

Comment:
I do not think the design of this package is reasonable. You named this package as `slice` or `[]`, but what I see is slice which has the type of string, or `[]string`. What you named and what you wish to do is quite different.

Therefore, I think you should rename the package to make it just correct and brings no ambiguousness.